session_srtcp.go

```diff
@@ -112,25 +112,47 @@ func (s *SessionSRTCP) decrypt(buf []byte) error {
 		return err
 	}
 
-	pkt, err := rtcp.Unmarshal(decrypted)
-	if err != nil {
-		return err
-	}
-
-	for _, ssrc := range pkt.DestinationSSRC() {
+	forward := func(ssrc uint32) (closed bool, err error) {
 		r, isNew := s.session.getOrCreateReadStream(ssrc, s, newReadStreamSRTCP)
 		if r == nil {
-			return nil // Session has been closed
+			return true, nil // Session has been closed
 		} else if isNew {
 			s.session.newStream <- r // Notify AcceptStream
 		}
 
 		readStream, ok := r.(*ReadStreamSRTCP)
 		if !ok {
-			return fmt.Errorf("failed to get/create ReadStreamSRTP")
+			return false, fmt.Errorf("failed to get/create ReadStreamSRTP")
 		}
 
 		_, err = readStream.write(decrypted)
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	}
+
+	pkt, err := rtcp.Unmarshal(decrypted)
+	if err != nil {
+		return err
+	}
+
+	cpkt := rtcp.CompoundPacket(pkt)
+	ssrcSet := make(map[uint32]bool)
+	if cpkt.Validate() == nil {
+		for _, ssrc := range cpkt.DestinationSSRC() {
+			ssrcSet[ssrc] = true
+		}
+	} else {
+		for _, p := range pkt {
+			for _, ssrc := range p.DestinationSSRC() {
```

Comment:
I wasn't sure what sssrcSet was for at first. Maybe this could be pulled out into a function?

```go
for _, ssrc := range destinationSSRCs(packets) {
  ...
}
```

Reply:
The first branch assumes that the DestinationSSRC for a whole CompoundPacket set applies to the whole set, the second branch creates a superset of all of the DestinationSSRC for each packet in the set

the behavior is slightly different (not sure if that's desireable)

Reply:
what about adding this function to the rtcp repository:

```
func DestinationSSRC(pkts []Packet) []uint32 {
	cpkt := CompoundPacket(pkts)
	if cpkt.Validate() == nil {
		//this is a CompoundPacket; the rules about where to send
		//it are pretty well-documented.
		return cpkt.DestinationSSRC()
	}

	ssrcSet := make(map[uint32]struct{})
	//this isn't a proper CompoundPacket; let's create a list of Destination SSRCs
	//that's a superset of all Destinations in the slice.
	for _, p := range pkts {
		for _, ssrc := range p.DestinationSSRC() {
			ssrcSet[ssrc] = struct{}{}
		}
	}

	out := make([]uint32, len(ssrcSet))

	for ssrc := range ssrcSet {
		out = append(out, ssrc)
	}

	return out
}
```

Reply:
I'd rather not grow the public API right now. If it's useful we can add it later.